content/articles/introduction-to-java-swing-desktop-application-using-netbeans-ide/index.md

```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Creates new form sectionHomeWindow
+     */
+    public sectionHomeWindow() {
+        
+    }
+}
+```
+
+#### JLabel
+JLabel component is an object component whose class displays a readable text or an image in the Swing Container User interface. The application user cannot change the text (edit) rendered in the JLabel however, the application itself, through action events, can change the text. The JLabel methods to display the text can allow both plain text and HTML texts rendered in the UI as plain text.
+
+Example;
+
+``` java
+JLabel labelName=new JLabel("Name");
+```
+
+![jlabel](/engineering-education/introduction-to-java-swing-desktop-application-using-netbeans-ide/jlabel.png)
+
+#### JTextField
+It is a text swing component that allows users to input one line of text. JTextField inherits from the JTextComponent class of javax.swing Library
+
+Example declaration;
+
+``` java
+JTextField sectionTextField=new JTextField(20);
+```
+
+![jtextfield](/engineering-education/introduction-to-java-swing-desktop-application-using-netbeans-ide/jtext.png)
+ 
+ For JTextField to be accessible and editable, enable property must be set to true. Text field can be initialized by calling the JTextField constructor and passing a parameter of integer type. The integer parameter type passed in the constructor does limit the number of characters a user and key in the text field box, but it sets the width of the text field box in the number of columns to be assigned to it.
+
+#### JButton
+JButton is one of the swing components which gives swing the property of platform independence. JButton gives this implementation. This component creates a push (On click) effect on the application user interface, resulting in some action behind the code on it. It is implemented in any application by calling any of its class constructors. The click or double click push on it results in retrieving the data from databases and displaying them on the UI or collecting the user's data on UI and storing/saving them into the database. JButton in most applications contains text or image display which communicates to the user what the button does. 
+
+Example Declaration; 
+
+``` java
+JButton submitButton=new JButton("Submit");
+```
+
+![jbutton](/engineering-education/introduction-to-java-swing-desktop-application-using-netbeans-ide/jbutton.png)
+
+### Getting Started with Java Swing desktop application in NetBeans-Sample Application
+NetBeans- is an IDE for developing an application in different languages. Java programming language of one of those many languages which it supports. NetBeans is the best Swing desktop application developing IDE since it provides a user-friendly interface for designing and coding UI. It has inbuilt libraries for swing, which auto-populate codes. The inbuilt libraries in NetBeans provide a palette section of the IDE when creating a swing application, making it easier to drag and drop and components for designing an application.
```

Comment:
```Suggestion
NetBeans is an IDE for developing applications using different languages. Java programming language is one of those many languages which NetBeans supports. NetBeans is the best IDE for building desktop applications using swing since it provides a user-friendly interface for designing and coding the UI. It has inbuilt libraries for swing, which auto-populate code. The inbuilt libraries in NetBeans provide a palette section in the IDE when creating a swing application, making it easier to drag and drop and components for designing an application.
```